dialect/sql/schema/postgres.go

```diff
@@ -332,15 +347,241 @@ func (d *Postgres) addColumn(c *Column) *sql.ColumnBuilder {
 // alterColumn returns list of ColumnBuilder for applying in order to alter a column.
 func (d *Postgres) alterColumn(c *Column) (ops []*sql.ColumnBuilder) {
 	b := sql.Dialect(dialect.Postgres)
-	ops = append(ops, b.Column(c.Name).Type(d.cType(c)))
+
+	// Migrating to enum? this will be invoked even when the column is already an enum
+	if typ, ok := c.SchemaType[dialect.Postgres]; ok && c.Type == field.TypeEnum {
+		ops = append(ops, b.Column(c.Name).Type(typ).Attr(fmt.Sprintf("using (%s::%s)", typ, typ)))
+	} else {
+		ops = append(ops, b.Column(c.Name).Type(d.cType(c)))
+	}
+
 	if c.Nullable {
 		ops = append(ops, b.Column(c.Name).Attr("DROP NOT NULL"))
 	} else {
 		ops = append(ops, b.Column(c.Name).Attr("SET NOT NULL"))
 	}
+
 	return ops
 }
 
+const nativeEnumsQuery = `SELECT pg_type.typname,
+       pg_enum.enumlabel
+FROM pg_type
+         JOIN
+     pg_enum ON pg_enum.enumtypid = pg_type.oid
+         INNER JOIN pg_namespace on pg_namespace.oid = pg_type.typnamespace
+WHERE pg_namespace.nspname = current_schema()
+ORDER BY pg_type.typname, pg_enum.enumlabel`
+
+func (d *Postgres) enumChanges(ctx context.Context, tx dialect.Tx, tables ...*Table) (*enumChanges, error) {
+	expectedEnums := make(map[string]*NativeEnum)
+
+	for _, t := range tables {
+		// How do we retrieve annotation here?
```

Comment:
We could alter behaviour using annotations but they do not seem to be copied from field to column